extensions/amp-form/0.1/amp-form.js

```diff
@@ -532,17 +579,24 @@ export class AmpForm {
 
   /**
    * Transition the form to the submit success state.
-   * @param {!../../../src/service/xhr-impl.FetchResponse} response
+   * @param {!../../../src/service/xhr-impl.FetchResponse|!Object} response
    * @return {!Promise}
    * @private visible for testing
    */
-  handleXhrSubmitSuccess_(response) {
-    return response.json().then(json => {
+  handleSubmitSuccess_(response) {
+    const isSsrSupported = this.ssrTemplateHelper_.isSupported();
+    const jsonPromise = isSsrSupported
+      ? Promise.resolve(response.renderedHtml) : response.json();
+
+    return jsonPromise.then(json => {
       this.triggerAction_(/* success */ true, json);
-      this.triggerFormSubmitInAnalytics_('amp-form-submit-success');
       this.setState_(FormState_.SUBMIT_SUCCESS);
       this.renderTemplate_(json || {});
-      this.maybeHandleRedirect_(response);
+
+      if (!isSsrSupported) {
```

Comment:
I'm a little confused, why don't we want to allow redirecting if the viewer can render templates? These form features seem unrelated to whether the viewer or the page itself render the form's templates.